### PR TITLE
[DOCS] Reviews Client helpers section in the Node.JS book

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR relocates the `Client helpers` section in the TOC of the Node.JS documentation to be in line with the proposed structure of the Clients docs. It also reviews the text of the page and adds IDs to the sections.

This PR is part of the Client docs redesign effort. Related issue: https://github.com/elastic/clients-team/issues/257

### Preview

[Node.JS TOC](https://elasticsearch-js_1434.docs-preview.app.elstc.co/guide/en/elasticsearch/client/javascript-api/master/index.html)

### CODE REVIEW

File: docs/helpers.asciidoc [hunk 1]
```diff
@@ -466,12 +490,16 @@ for (const doc of documents) {
...
+[[scroll-search-helper]]
+=== Scroll search helper
 
 ~Added~ ~in~ ~`v7.7.0`~
 
-This helpers offers a simple and intuitive way to use the scroll search API. Once called, it returns an https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of[async iterator] which can be used in conjuction with a for-await...of. +
-It handles automatically the `429` error and uses the client's `maxRetries` option.
+This helpers offers a simple and intuitive way to use the scroll search API. 
+Once called, it returns an 
+https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of[async iterator]
```
Comment (jamesorlakin) [inline docs/helpers.asciidoc:500]: I've noticed this link gets encoded strangely. The three dots get substituted with an ellipsis which then results in a 404: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await%E2%80%A6%E2%80%8Bof

My knowledge of Asciidoc is quite low unfortunately, but would the [`link:` macro](https://docs.asciidoctor.org/asciidoc/latest/macros/link-macro/) resolve it?

Comment (szabosteve) [reply docs/helpers.asciidoc]: Thanks for spotting! Unfortunately, `link:macro` won't resolve it correctly. I change the URL to the URL of the async function page as link text refers to that one anyway.

**Review by delvedor — APPROVED** [review]
LGTM

File: docs/index.asciidoc [hunk 2]
```diff
@@ -19,3 +19,4 @@ include::typescript.asciidoc[]
 include::reference.asciidoc[]
 include::examples/index.asciidoc[]
 include::helpers.asciidoc[]
+include::breaking-changes.asciidoc[]
```
Comment (szabosteve) [inline docs/index.asciidoc:22]:
```suggestion
```

**Comment by szabosteve** [comment]
@elasticmachine test this please

**Pull request merged**